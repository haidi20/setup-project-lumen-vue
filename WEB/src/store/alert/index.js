// partials
import alertStates from './_partials/alertStates'
import alertGetters from './_partials/alertGetters'
import alertMutations from './_partials/alertMutations'
import alertActions from './_partials/alertActions'

const alert = {
    namespaced: true,
    state: alertStates,
    getters: alertGetters,
    actions: alertActions,
    mutations: alertMutations,
}

export default alert