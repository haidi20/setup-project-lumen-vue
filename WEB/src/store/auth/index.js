import authStates from './authStates';
import authGetters from './authGetters';
import authActions from './authActions';
import authMutations from './authMutations';

const auth = {
    namespaced: true,
    state: authStates,
    getters: authGetters,
    actions: authActions,
    mutations: authMutations,
}

export default auth