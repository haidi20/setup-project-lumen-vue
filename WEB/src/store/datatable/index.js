import http from '../../api'

const datatable = {
    namespaced: true,
    state: {
        data: [],
        headers: [
            { text: "No.", sortable: false, value: "number", class: "bg-header-info" },
            { text: "Actions", value: "actions", sortable: false, align: "start", class: "bg-header-info" },
            { text: "Title", sortable: false, value: "title", class: "bg-header-info" },
            { text: "Date", value: "date", sortable: false, class: "bg-header-info" },
            { text: "Author", value: "author", sortable: false, class: "bg-header-info" },
        ],
        page: 1,
        totalPages: 0,
        pageSize: 5,
        pageSizes: [5, 10],
        loading: false,
        search: '',
        awaitingSearch: false,
    },
    getters: {
        getNumberData: state => item => {
            let number = state.data.map(function(x) { return x.id; }).indexOf(item.id) + 1;

            if (state.page == 1) {
                return number;
            } else {
                return (number + state.pageSize);
            }
        },
    },
    mutations: {
        NULL_DATA: (state) => {
            state.data = [];
        },
        INSERT_DATA: (state, payload) => {
            state.data = payload.data;
            state.totalPages = payload.last_page;
        },
        LOADING_TRUE: (state) => {
            state.loading = true;
        },
        LOADING_FALSE: (state) => {
            state.loading = false;
        },
    },
    actions: {
        async fetchData({ commit }, payload) {
            // const params = {
            //     page: state.page,
            //     search: state.search,
            //     per_page: state.pageSize,
            // }
            commit('NULL_DATA');
            commit('LOADING_TRUE');
            try {
                await http.get(`/api${payload.dataLink}`)
                    .then(ress => {
                        let data = ress.data.data;
                        commit('INSERT_DATA', data);
                        commit('LOADING_FALSE');
                    });
            } catch (error) {
                console.log('error fetch data = ' + error);
                return (null);
            }
        },
        // PageChange({ commit }) {
        //     // state.page = value;
        //     // this.fetchData();
        // },
        // PageSizeChange({ commit }) {
        //     // state.page = 1;
        //     // state.pageSize = size;
        //     // state.fetchData();
        // },
        // watchSearch: function({ commit }) {
        //     // if (!state.awaitingSearch) {
        //     //     setTimeout(() => {
        //     //         // this.fetchData();
        //     //         state.awaitingSearch = false;
        //     //     }, 2000); // 2 sec delay
        //     // }

        //     // state.awaitingSearch = true;
        // },
        // save_donasi({ commit, rootState }, payload) {
        //     // rootState BERARTI MENGAKSES STATE YANG TIDAK BERADA DALAM MODULES
        //     // DALAM HAL INI STATE isLoading YANG ADA DI DALAM FILE store.js
        //     rootState.isLoading = true //SET TRUE UNTUK MEMBERIKAN EFEK LOADING
        //     setTimeout(() => {
        //         //MENGINSTRUKSIKAN PADA MUTATIONS TERKAIT UNTUK MENJALANKAN INSTRUKSINYA
        //         commit('ADD_DONASI', payload)
        //         // STATE isLoading DI MATIKAN KEMBALI
        //         rootState.isLoading = false
        //     }, 1000)
        // }
    }
}

export default datatable