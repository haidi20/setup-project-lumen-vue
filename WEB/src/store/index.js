import Vue from 'vue'
import Vuex from 'vuex'

import auth from './auth'
import alert from './alert'
import datatable from './datatable'

Vue.use(Vuex)

export default new Vuex.Store({
    modules: {
        auth,
        alert,
        datatable,
    }
});