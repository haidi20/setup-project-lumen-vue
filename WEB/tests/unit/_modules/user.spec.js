import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import { mount, createLocalVue } from '@vue/test-utils';

//components
import User from '@/components/user';

import store from '@/store';

Vue.use(Vuex);
Vue.use(Vuetify);

describe('User', () => {
    const localVue = createLocalVue()
    let vuetify, wrapper;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(User, {
            localVue,
            vuetify,
            store,
        });
    })

    it('snapshot page user', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('check word title', async() => {
        const title = wrapper.find('h2');
        expect(title.text()).toBe('User');
    });
});