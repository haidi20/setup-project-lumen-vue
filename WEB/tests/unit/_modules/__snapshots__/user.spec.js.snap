// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User snapshot page user 1`] = `
<div>
  <div>
    <div>
      <h2>User</h2>
      <p> </p>
    </div>
  </div>
</div>
`;
